### PR TITLE
Expose sqlite open flags

### PR DESCRIPTION
**Pull request by mrmacete**
Example: `SqliteDatabase.open(fileName, {flags: [‘readwrite’, ‘create’]})`

Possible flags are `readonly`, `readwrite`, `create`. The `create` flag can be combined only with `readwrite`. The default flags are `[‘readwrite’, ‘create’]`.

### CODE REVIEW

File: bindings/gumjs/gumv8database.cpp [hunk 1]
```diff
@@ -157,16 +157,16 @@ _gum_v8_database_finalize (GumV8Database * self)
 GUMJS_DEFINE_FUNCTION (gumjs_database_open)
 {
   gchar * path;
+  gint flags;
   sqlite3 * handle;
   gint status;
   Local<Object> object;
 
-  if (!_gum_v8_args_parse (args, "s", &path))
+  if (!_gum_v8_args_parse (args, "si", &path, &flags))
```
Comment (oleavr) [inline bindings/gumjs/gumv8database.cpp]: I guess `guint` / `'u'` would be more appropriate?

File: bindings/gumjs/gumdukdatabase.c [hunk 2]
```diff
@@ -138,16 +138,16 @@ GUMJS_DEFINE_FUNCTION (gumjs_database_open)
...
   GumDukDatabase * self;
   const gchar * path;
+  gint flags;
   sqlite3 * handle;
   gint status;
 
   self = gumjs_module_from_args (args);
 
-  _gum_duk_args_parse (args, "s", &path);
+  _gum_duk_args_parse (args, "si", &path, &flags);
```
Comment (oleavr) [inline bindings/gumjs/gumdukdatabase.c]: I guess `guint` / `'u'` would be more appropriate?

File: bindings/gumjs/gumv8database.cpp (hunk 1 above)
Comment (mrmacete) [reply bindings/gumjs/gumv8database.cpp]: well, i thought about that too... but from the docs it seems `sqlite3_open_v2` consumes signed flags: https://sqlite.org/c3ref/open.html

File: bindings/gumjs/runtime/core.js [hunk 3]
```diff
@@ -762,4 +762,43 @@ SourceMap.prototype.resolve = function (generatedPosition) {
...
+  readwrite: 2,
+  create: 4,
+};
+
+Object.defineProperties(SqliteDatabase, {
+  open: {
+    enumerable: true,
+    value: function (file, options) {
+      if (options === undefined)
+        options = {};
```
[the remark this answers is not in the document]
Comment (mrmacete) [reply bindings/gumjs/runtime/core.js]: awesome!

File: bindings/gumjs/gumv8database.cpp (hunk 1 above)
Comment (oleavr) [reply bindings/gumjs/gumv8database.cpp]: Ahh, nice catch!

**Pull request merged**